src/main/scala/temple/test/internal/EndpointTest.scala

```diff
@@ -1,10 +1,98 @@
+        assert(responseForKey.isBoolean, s"$key is not of type bool")
+      case AttributeType.DateType =>
+        val stringValue = responseForKey.asString.getOrElse(fail(s"$key is not a valid string"))
+        assert(Try(new SimpleDateFormat("yyyy-MM-dd").parse(stringValue)).isSuccess, s"$key is not a valid date")
+      case AttributeType.DateTimeType =>
+        val stringValue = responseForKey.asString.getOrElse(fail(s"$key is not a valid string"))
+        assert(
+          Try(new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX").parse(stringValue)).isSuccess,
+          s"$key is not a valid datetime",
+        )
+      case AttributeType.TimeType =>
+        val stringValue = responseForKey.asString.getOrElse(fail(s"$key is not a valid string"))
+        assert(Try(new SimpleDateFormat("HH:mm:ssXXX").parse(stringValue)).isSuccess, s"$key is not a valid time")
+      case AttributeType.BlobType(_) =>
+        assert(responseForKey.isString, s"$key is not a valid string")
+      case AttributeType.StringType(max, min) =>
+        val stringValue = responseForKey.asString.getOrElse(fail(s"$key is not a valid string"))
+        min.foreach { minValue =>
+          assert(stringValue.length >= minValue, s"$key does not satisfy min bound of $minValue")
+        }
+        max.foreach { maxValue =>
+          assert(stringValue.length <= maxValue, s"$key does not satisfy max bound of $maxValue")
+        }
+      case AttributeType.IntType(max, min, _) =>
+        // TODO: precision
+        val intValue = responseForKey.asNumber.flatMap(_.toInt).getOrElse(fail(s"$key is not a valid number"))
+        min.foreach { minValue =>
+          assert(intValue >= minValue, s"$key does not satisfy min bound of $minValue")
+        }
+        max.foreach { maxValue =>
+          assert(intValue <= maxValue, s"$key does not satisfy max bound of $maxValue")
+        }
+      case AttributeType.FloatType(max, min, _) =>
+        // TODO: precision
+        val floatValue = responseForKey.asNumber.map(_.toFloat).getOrElse(fail(s"$key is not a valid float"))
+        min.foreach { minValue =>
+          assert(floatValue >= minValue, s"$key does not satisfy min bound of $minValue")
+        }
+        max.foreach { maxValue =>
+          assert(floatValue <= maxValue, s"$key does not satisfy max bound of $maxValue")
+        }
+    }
+
+  def validateResponseBody(request: JsonObject, response: JsonObject, attributes: Map[String, Attribute]): Unit = {
+    // The response should contain exactly the keys in attributes, PLUS an ID attribute, MINUS anything that is @server
+    val expectedAttributes = attributes.filter {
+      case (_, attribute) =>
+        attribute.accessAnnotation.fold(true) {
+          case Annotation.Server                        => false
+          case Annotation.ServerSet | Annotation.Client => true
+        }
+    }
+    val expectedAttributeNames = (expectedAttributes.keys ++ Seq("ID")).map(_.capitalize).toSet
```

Comment:
Flip the args of the ++, replace Seq with Set, and there’s no need for the toSet.